### PR TITLE
Spike/sc 1517

### PR DESCRIPTION
**Pull request by abnepc**
## Jira Story
[SC-1517](https://epcpower.atlassian.net/browse/SC-1517)
## About
Investigate/remove `sunspecdemo` from the repo

The `sunspec-demo` python dependency was removed. The purpose of the removal is so the software team has one less repository to maintain. The `sunspec-demo` repository was unnecessarily the dependency of several repositories. The small amount of code that was being executed in `sunspec-demo` is now in `stlib`.
As a bonus, included in this removal are updated application naming for the M repos. For example:
* `pm` -> `mpm` (or `epcpm` -> `mpm`)
* `grid-tied`/`gridtied` -> `mtcu` or `mbcu`
* many directory names and file names have similarly changed
* some environment variables have changed (ex: `GRIDTIED_*` -> `MTCU_*`)

## Associated Pull Requests
* [stlib](https://github.com/epcpower/stlib/pull/48)
* [st](https://github.com/epcpower/st/pull/49)
* [pm](https://github.com/epcpower/pm/pull/93)
* [grid-tied](https://github.com/epcpower/grid-tied/pull/683)
* [rm-dc](https://github.com/epcpower/rm-dc/pull/21)
* [m-bcu](https://github.com/epcpower/m-bcu/pull/7)
* [m-tcu](https://github.com/epcpower/m-tcu/pull/53)
## Reproduction Steps
N/A
## Validation
Delete your `.venv` folder and do a `poetry install`. Then make sure that `sunspecdemo` was not installed. 

[SC-1517]: https://epcpower.atlassian.net/browse/SC-1517?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "conventionalCommits.scopes": ["SC-1517"]
```
Comment (epcAnker) [inline .vscode/settings.json]: remove file, gitignore

**Review by epcAnker — CHANGES_REQUESTED** [review]
Remove MPM_ folder from PR

"epcpm" changes needed in:
README.rst
.gitattributes
src/mpm/__main__.py "    file_handler = logging.FileHandler("epcpm.log")"

Look into:
epcpmcli_Linux_xml (and other xml files)
*.py files with "epcpm" they are all commented


Would also recommend looking at *.py files

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
-name = "epcpm"
-version = "0.0.0"
+name = "mpm"
+version = "0.0.0.post1176.dev0+6680080"
```
Comment (epcAnker) [inline pyproject.toml]: revert back to 0.0.0

File: src/mpm/__build.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# This file has been generated
```
Comment (epcAnker) [inline src/mpm/__build.py]: remove generated file

File: src/mpm/__init__.py [hunk 4]
```diff
@@ -10,8 +10,8 @@ def get_git_revision_hash() -> str:
 # to the build placeholder "0.0.0". (In other words: leave alone!)
 # For release/distribution, the __version__ variable below is modified
 # during CI by poetry dynamic versioning with the github tagged version.
-__version__ = "0.0.0"
+__version__ = "0.0.0.post1176.dev0+6680080"
```
Comment (epcAnker) [inline src/mpm/__init__.py]: set back to 0.0.0

File: src/mpm/__version.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+# This file was automatically generated by sub/epyqlib/epyqlib/cli/generateversion.py
```
Comment (epcAnker) [inline src/mpm/__version.py]: remove generated file

File: pyproject.toml [hunk 6]
```diff
@@ -68,23 +61,23 @@ pytest-twisted = "<=1.13.4"
...
-epcexportdocx = 'epcpm.cli.exportdocx:cli'
-epcconvertparameters = 'epcpm.cli.convertepp:cli'
-builduipm = 'epcpm.builduipm:compile_ui'
+mpmcli = 'mpm.cli.main:main'
+epcparameterstoc = 'mpm.cli.parameterstoc:cli'
+epcimportsym = 'mpm.cli.importsym:cli'
+epcexportsym = 'mpm.cli.exportsym:cli'
+epcexportdocx = 'mpm.cli.exportdocx:cli'
+epcconvertparameters = 'mpm.cli.convertepp:cli'
+builduipm = 'mpm.builduipm:compile_ui'
```
Comment (epcAnker) [inline pyproject.toml]: @gordon-epc do we want to change builduipm-builduimpm?

File: src/mpm/builduipm.py [hunk 7]
```diff
@@ -3,7 +3,7 @@
```
Comment (epcAnker) [inline src/mpm/builduipm.py]: @gordon-epc do we want to change builduipm->buildui**mpm**?

File: src/mpm/cantoxlsx.py [hunk 8]
```diff
@@ -9,7 +9,7 @@
 import typing
 import glob
 import uuid
-import epcpm.pm_helper
+import mpm.pm_helper
```
Comment (epcAnker) [inline src/mpm/cantoxlsx.py]: @gordon-epc do we want pm\_helper -> mpm\_helper???

File: src/mpm/tests/test_canmodel_parametermodel.json
Comment (epcAnker) [inline src/mpm/tests/test_canmodel_parametermodel.json:1]: can likely delete this file?

**Review by epcAnker — CHANGES_REQUESTED** [review]

File: MPM__/Pint-0.11.dist-info/AUTHORS
Comment (gordon-epc) [inline MPM__/Pint-0.11.dist-info/AUTHORS:1]: This whole `MPM__` directory should not be checked in. Where did it come from?

**Review by gordon-epc — CHANGES_REQUESTED** [review]

File: src/mpm/builduipm.py (hunk 7 above)
Comment (gordon-epc) [reply src/mpm/builduipm.py]: All or nothing.

File: src/mpm/cantoxlsx.py (hunk 8 above)
Comment (gordon-epc) [reply src/mpm/cantoxlsx.py]: All or nothing.

File: MPM__/Pint-0.11.dist-info/AUTHORS
Comment (abnepc) [reply MPM__/Pint-0.11.dist-info/AUTHORS:1]: [f49d990](https://github.com/epcpower/m-pm/pull/4/commits/f49d990f1b0aa7aa9379d79f1c85656c231abbae)

File: .vscode/settings.json (hunk 1 above)
Comment (abnepc) [reply .vscode/settings.json]: [ad51c84](https://github.com/epcpower/m-pm/pull/4/commits/ad51c84889958c333709fa33d6fefebf35dcd131) and [3ab51e1](https://github.com/epcpower/m-pm/pull/4/commits/3ab51e1be9e2efb3054dce62ab423dc1d0a8719c)

File: src/mpm/__version.py (hunk 5 above)
Comment (abnepc) [reply src/mpm/__version.py]: [49542d1](https://github.com/epcpower/m-pm/pull/4/commits/49542d19bbe919340d7f070f1a2ea1312f23ed93)

File: src/mpm/__build.py (hunk 3 above)
Comment (abnepc) [reply src/mpm/__build.py]: [49542d1](https://github.com/epcpower/m-pm/pull/4/commits/49542d19bbe919340d7f070f1a2ea1312f23ed93)

File: src/mpm/__init__.py (hunk 4 above)
Comment (abnepc) [reply src/mpm/__init__.py]: [52420a3](https://github.com/epcpower/m-pm/pull/4/commits/52420a389ef01663be40abb3e52637d4c26c4b7b)

File: pyproject.toml (hunk 2 above)
Comment (abnepc) [reply pyproject.toml]: [52420a3](https://github.com/epcpower/m-pm/pull/4/commits/52420a389ef01663be40abb3e52637d4c26c4b7b)

File: src/mpm/cantoxlsx.py (hunk 8 above)
Comment (abnepc) [reply src/mpm/cantoxlsx.py]: [ac96c9f](https://github.com/epcpower/m-pm/pull/4/commits/ac96c9f72ad23da10b920515bcea1aa3bf8b6d84)

File: src/mpm/builduipm.py (hunk 7 above)
Comment (abnepc) [reply src/mpm/builduipm.py]: [ac96c9f](https://github.com/epcpower/m-pm/pull/4/commits/ac96c9f72ad23da10b920515bcea1aa3bf8b6d84)

File: src/mpm/tests/test_canmodel_parametermodel.json
Comment (abnepc) [reply src/mpm/tests/test_canmodel_parametermodel.json:1]: [cb38c74](https://github.com/epcpower/m-pm/pull/4/commits/cb38c744cd8c9d42256c751e57fc371f2ee9c4fc)

File: pyproject.toml (hunk 6 above)
Comment (abnepc) [reply pyproject.toml]: [ac96c9f](https://github.com/epcpower/m-pm/pull/4/commits/ac96c9f72ad23da10b920515bcea1aa3bf8b6d84)

**Comment by abnepc** [comment]
> Remove MPM_ folder from PR
> 
> "epcpm" changes needed in: README.rst .gitattributes src/mpm/**main**.py " file_handler = logging.FileHandler("epcpm.log")"
> 
> Look into: epcpmcli_Linux_xml (and other xml files) *.py files with "epcpm" they are all commented
> 
> Would also recommend looking at *.py files

[ce80509](https://github.com/epcpower/m-pm/pull/4/commits/ce80509fb97d396436cfcd70432cf612227db589) and [fc124d5](https://github.com/epcpower/m-pm/pull/4/commits/fc124d55b963944fdbd36f31076c6e3b6b6eddd7). As for files in `.idea` directory, those should've been in a `.gitignore`. I have an adjacent PR that adds to the `.gitignore` and can add it there if you'd prefer?

**Review by vvainola — APPROVED** [review]

**Review by gordon-epc — APPROVED** [review]

**Review by epcAnker — APPROVED** [review]

**Pull request merged**